ui/org.eclipse.pde.core/src/org/eclipse/pde/internal/core/target/P2TargetUtils.java

```diff
@@ -984,44 +985,108 @@ private static <T> T getP2Service(Class<T> key, String absentErrorMessage) throw
 	/**
 	 * Return a queryable on the metadata defined in the given repo locations
 	 *
-	 * @param repos the repos to lookup
-	 * @param followRepositoryReferences whether to follow repository references
-	 * @param monitor the progress monitor
+	 * @param repos
+	 *            the repos to lookup
+	 * @param followRepositoryReferences
+	 *            whether to follow repository references
+	 * @param monitor
+	 *            the progress monitor
 	 * @return the set of metadata repositories found
-	 * @throws CoreException if there is a problem getting the repositories
+	 * @throws CoreException
+	 *             if there is a problem getting the repositories
 	 */
 	static IQueryable<IInstallableUnit> getQueryableMetadata(Collection<URI> repos, boolean followRepositoryReferences,
 			IProgressMonitor monitor) throws CoreException {
-		IMetadataRepositoryManager manager = getRepoManager();
+		return getQueryableMetadata(repos, followRepositoryReferences, false, monitor);
+	}
+
+	/**
+	 * Return a queryable on the metadata defined in the given repo locations
+	 *
+	 * @param repos
+	 *            the repos to lookup
+	 * @param followRepositoryReferences
+	 *            whether to follow repository references
+	 * @param forceReload
+	 *            forces reloading of repositories
+	 * @param monitor
+	 *            the progress monitor
+	 * @return the set of metadata repositories found
+	 * @throws CoreException
+	 *             if there is a problem getting the repositories
+	 */
+	static IQueryable<IInstallableUnit> getQueryableMetadata(Collection<URI> repos, boolean followRepositoryReferences,
+			boolean forceReload,
+			IProgressMonitor monitor) throws CoreException {
+		IMetadataRepositoryManager metadataRepositoryManager = getRepoManager();
+		Collection<URI> existing = new LinkedHashSet<>(
+				Arrays.asList(metadataRepositoryManager.getKnownRepositories(IRepositoryManager.REPOSITORIES_ALL)));
 		if (repos.isEmpty()) {
-			repos = Arrays.asList(manager.getKnownRepositories(IRepositoryManager.REPOSITORIES_ALL));
+			repos = existing;
 		}
-		SubMonitor subMonitor = SubMonitor.convert(monitor, repos.size() * 2);
-
+		int work = repos.size() * (forceReload?4: 2);
```

Comment:
Does the formatter not add white space around the `?`? 
```suggestion
		int work = repos.size() * (forceReload ?4  2);
```